File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/base/TypingAndDedupingFlag.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.base;

import java.util.Optional;

public class TypingAndDedupingFlag {

  public static boolean isDestinationV2() {
    return DestinationConfig.getInstance().getIsV2Destination()
        || DestinationConfig.getInstance().getBooleanValue("use_1s1t_format");
  }

  public static Optional<String> getRawNamespaceOverride(final String option) {
    final String rawOverride = DestinationConfig.getInstance().getTextValue(option);
    if (rawOverride == null || rawOverride.isEmpty()) {
      return Optional.empty();
    } else {
      return Optional.of(rawOverride);
    }
  }

}

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/TypingAndDedupingFlag.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.base

import java.util.*

object TypingAndDedupingFlag {
    @JvmStatic
    val isDestinationV2: Boolean
        get() =
            (DestinationConfig.Companion.instance!!.isV2Destination ||
                DestinationConfig.Companion.instance!!.getBooleanValue("use_1s1t_format"))

    @JvmStatic
    fun getRawNamespaceOverride(option: String?): Optional<String> {
        val rawOverride: String = DestinationConfig.Companion.instance!!.getTextValue(option)
        return if (rawOverride.isEmpty()) {
            Optional.empty()
        } else {
            Optional.ofNullable(rawOverride)
        }
    }
}
